### ISSUE TITLE
builtins.fetchTree requires flakes

### ISSUE THREAD

**Issue by samueldr** [NixOS/nix#5541]
**Describe the bug**

Going from the release notes, `builtins.fetchTree` isn't described as requiring flakes to be enabled.

> `builtins.fetchTree` allows fetching a source tree using any backends supported by the fetcher infrastructure. It subsumes the functionality of existing built-ins like `fetchGit`, `fetchMercurial` and `fetchTarball`.

**Steps To Reproduce**

```
[nix-shell:~/tmp/nix]$ ./outputs/out/bin/nix --extra-experimental-features nix-command repl
Welcome to Nix 2.5. Type :? for help.

nix-repl> builtins.fetchTree {}
error: experimental Nix feature 'flakes' is disabled; use '--extra-experimental-features flakes' to override
```

**Expected behavior**

`builtins.fetchTree` to be available even without flakes being enabled.

**`nix-env --version` output**

Built from 9d5c985aedcf7d11a5310eae6f27dcde8eb7f608

**Additional context**

This may be a documentation bug. But I suspect it shouldn't be.

**Comment by roberth** [comment]
Maybe it's because of the reproducibility issues that it's gated, but then it should be a different feature flag like `unstable-fetchers` or something.

**Comment by samueldr** [comment]
@edolstra any input on that?

**Comment by AleXoundOS** [comment]
Yes, it is highly desirable to have `builtins.fetchTree` without flakes requirement or additional toggles. For example, companies that have local nixpkgs repository with ssh access need `shallow` clone feature to avoid downloading and storing 1.8 GiB with `builtins.fetchGit` (though, [`shallow` feature is currently broken](https://github.com/NixOS/nix/issues/5119)).

**Comment by fricklerhandwerk** [comment]
Discussed in Nix team meeting 2022-12-09:

- assigned to @edolstra to add documentation

**Comment by roberth** [comment]
This could take advantage of
 - #6314, and
 - #8371
to document that it's experimental.

**Comment by MattSturgeon** [comment]
I noticed the manual documents a `fetch-tree` experimental feature, and #15654 fixes an issue where URL fetches don't work without the `flakes` experimental feature.

I have a similar example:
```
$ nix --version
nix (Nix) 2.34.6

$ nix-instantiate --option experimental-features fetch-tree --eval --expr '
    fetchTree {
      type = "github";
      owner = "NixOS";
      repo = "nixpkgs";
      rev = "d99b013d5d1931ad77fe3912ed218170dec5d9a4";
      narHash = "sha256-7DKWmH23hL2eYdkxCKeqj2i+yljTKuU+3Nk1UPHOnxc=";
      lastModified = 1779259093;
    }
  '
error:
       … while calling the 'fetchTree' builtin
         at «string»:1:1:
            1| fetchTree {
             | ^
            2|   type = "github";

       error: experimental Nix feature 'flakes' is disabled; add '--extra-experimental-features flakes' to enable it
```

`fetch-tree` enables the `fetchTree` builtin, but I still cannot actually _invoke_ the builtin without `flakes`.

I'm not sure whether #15654 addresses this specific example or not?
